kratos/expression/variable_expression_data_io.cpp

```diff
@@ -48,6 +48,39 @@ void Assign(
     (rValue.insert_element(TIndex, rExpression.Evaluate(EntityIndex, EntityDataBeginIndex, TIndex)), ...);
 }
 
+template<class TDataType, std::size_t... TIndex>
+void ReadVector(
+    LiteralFlatExpression<std::conditional_t<std::is_same_v<typename TDataType::value_type, int>, int, double>>& rExpression,
+    const IndexType EntityDataBeginIndex,
+    const TDataType& rValue,
+    std::index_sequence<TIndex...>)
+{
+    IndexType local_index = 0;
+    for (IndexType i = 0; i < rValue.size(); ++i) {
+        (rExpression.SetData(EntityDataBeginIndex, local_index++, rValue[i][TIndex]), ...);
+    }
+}
+
+template<class TDataType, std::size_t... TIndex>
+void AssignVector(
+    const Expression& rExpression,
+    const IndexType EntityIndex,
+    const IndexType EntityDataBeginIndex,
+    TDataType& rValue,
+    std::index_sequence<TIndex...>)
+{
+    const auto& r_shape = rExpression.GetItemShape();
```

Comment:
check for degenerate shapes
```suggestion
    const auto& r_shape = rExpression.GetItemShape();
    if (r_shape.empty()) {
        rValue = TDataType();
        return;
    }
```

Reply:
I would not do it like that, since this will be called in inner most loops. And the shape should be checked before calling these methods. Just in case if someone misses, I have added a full debug error and a comment explaning the shapes should be checked.